### PR TITLE
JSONSchema: Add TupleLiteral support

### LINKED ISSUE #2460: Json schema emitter crash for tuples
[Playground Link](https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAdHlwZXNwZWMvanNvbi1zY2hlbWEiOwoKdXNpbmcgVHlwZVNwZWMuSnNvblPFHTsKCkDELMYOCm5hbWVzcGFjZSDGEXM7Cgptb2RlbCBGT28gewogIGE6IFtzdHJpbmcsIGludDMyXQp9&e=%40typespec%2Fjson-schema&options=%7B%7D)

```
ExternalError: Emitter "@typespec/json-schema" crashed! This is a bug. Please file an issue at https://github.com/microsoft/typespec/issues Error: TypeEmitter doesn't have a method named tupleLiteralValuesContext 
at compilerAssert (https://cadlplayground.z22.web.core.windows.net/libs/@typespec/compiler/emitter-utils-6402b6df.js:360:11) 
at setContextForType (https://cadlplayground.z22.web.core.windows.net/libs/@typespec/compiler/emitter-utils-6402b6df.js:21659:17) at withTypeContext (https://cadlplayground.z22.web.core.windows.net/libs/@typespec/compiler/emitter-utils-6402b6df.js:21691:9) 
```

### PR DESCRIPTION
**Pull request by tjprescott**
Fix #2460.

### CODE REVIEW

File: packages/compiler/src/emitter-framework/type-emitter.ts [hunk 1]
```diff
@@ -666,6 +666,14 @@ export class TypeEmitter<T, TOptions extends object = Record<string, never>> {
     return this.emitter.result.none();
   }
 
+  tupleLiteralValuesContext(tuple: Tuple): Context {
```
Comment (timotheeguerin) [inline packages/compiler/src/emitter-framework/type-emitter.ts:669]: I think you can remove the duplicate overide in `CodeTypeEmitter` now

File: packages/json-schema/test/tuples.test.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    deepStrictEqual(schemas, {
+      "Foo.json": {
+        $id: "Foo.json",
+        $schema: "https://json-schema.org/draft/2020-12/schema",
+        type: "object",
+        required: ["a"],
+        properties: {
+          a: {
+            type: "array",
+            items: [
```
Comment (timotheeguerin) [inline packages/json-schema/test/tuples.test.ts:20]: I don't think this is correct, `items` is just an object with the array type.

Tuple is done with prefixItems i think 

https://json-schema.org/understanding-json-schema/reference/array#tupleValidation

Comment (tjprescott) [reply packages/json-schema/test/tuples.test.ts:20]: Interesting, what I found online said it was an array of schemas. However, that says it's for Draft 4 and 6, and we are on draft 2020-12, which is described here: https://json-schema.org/draft/2020-12/json-schema-core#section-10.3.1.1

It's not entirely clear to me what it should be... @bterlson what say you?

Comment (tjprescott) [reply packages/json-schema/test/tuples.test.ts]: Updated it to prefixItems

Comment (bterlson) [reply packages/json-schema/test/tuples.test.ts]: The schemas for the tuple entries should be `prefixItems`. We had discussed and rejected adding any other length constraints.

Comment (tjprescott) [reply packages/json-schema/test/tuples.test.ts]: The length constraints just "happened" as a result of emitting the int type. Is there a way to suppress that for tuples or should it not be present on any int types?

Comment (bterlson) [reply packages/json-schema/test/tuples.test.ts]: Min/max on tuple items is fine, was talking about limiting the length of the array schema. The output looks good!

File: packages/compiler/src/emitter-framework/type-emitter.ts (hunk 1 above)
Comment (tjprescott) [reply packages/compiler/src/emitter-framework/type-emitter.ts:669]: Removed.

**Comment by tjprescott** [comment]
@bterlson @timotheeguerin comments addressed.

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**